### DISCUSSION TITLE
Update import on `<script src=>` when file is moved

### DISCUSSION THREAD

**Discussion by alexrecuenco** [withastro/language-tools#891] [Feature requests]
### Background & Motivation

It is a core feature of Astro [that script tags allow src local files](https://docs.astro.build/en/guides/client-side-scripts/#import-local-scripts)

It would be lovely if imports update when you update their location also in those files, similarly to how `es module` imports update.

### Open Questions

- How would it distinguish when the script is global but relative, vs when it is not. Would it require parsing and finding the is:global tag?

### Help make it happen!

- [ ] I am willing to submit a PR to implement this change.
- [X] I am willing to submit a PR to implement this change, but would need some guidance.
- [ ] I am not willing to submit a PR to implement this change.

**Comment by Princesseuh** [comment]
That would be quite cool! I'd be happy to help you implement this. I don't think quite remember which of the hooks is triggered on file changes, I believe it might be [`provideRenameEdits`](https://volarjs.dev/reference/services/#providerenameedits), but not 100% sure. Looking at the Volar service for TypeScript might be an useful reference: https://github.com/volarjs/services/tree/master/packages/typescript

Happy to provide more in-depth explanations once I'm back from holidays next week.

**Reply by alexrecuenco** [reply]
Awesome, I currently am struggling to find time, but I'd let you know as soon as I start trying this a bit, it shouldn't be that hard... I hope
